src/cmd/completions.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var completionsNoDesc bool
+
+var completionsCmd = &cobra.Command{
+	Use:   "completions [bash|zsh|fish|powershell]",
+	Short: "Generates auto-completion scripts for Bash, zsh, fish, and PowerShell",
+	Long: `Generates auto-completion scripts for Bash, zsh, fish, and PowerShell
+
+With completions enabled, git-town TAB will show you all possible
+subcommands. As a bonus, some shells even show the short help text next to it.
+
+To enable completions:
+
+Bash:
+
+$ source <(git-town completions bash)
+
+Persist and autoload on each session:
+
+Linux: $ git-town completions bash > /etc/bash_completion.d/git-town
+
+MacOS: $ git-town completions bash > /usr/local/etc/bash_completion.d/git-town
+
+Zsh:
+
+$ source <(git-town completions zsh)
+
+Persist and autoload on each session:
+
+$ git-town completions zsh > /usr/share/zsh/vendor-completions/_git-town
+
+Fish:
+
+$ git-town completions fish | source
+
+Persist and autoload on each session:
+
+$ git-town completions fish > /etc/fish/completions/git-town.fish
```

Comment:
Your friendly QA assistant @kevgo did some testing! The Bash autocompletion works! Fish autocompletion errors out with this: 

```
set: Variable name '__git-town_comp_commandLine' is not valid. See `help identifiers`.

~/.config/fish/completions/git-town.fish (line 60): 
        set __git-town_comp_commandLine (commandline)
        ^
in function '__git-town_prepare_completions'
in command substitution
```

Variable names in Fish cannot contain `-`. If I change the `-` to `_` in the variables (and only here), it works. This seems an upstream issue. We could replace all occurrences of `/__git-town_.*\b/` with that string that has `-` replaced with `_` as a workaround for now.

Reply:
oh... :D

Reply:
fortunately, this might be a short lived problem: https://github.com/spf13/cobra/pull/1122

Reply:
since judging by average this should be fixed in a few days, I'd rather say we refrain from a workaround now and trade some work for patience... Or we risk exposing the fish users and do a comment in the instructions meanwhile...

Reply:
@kevgo If you are ok with the cuationary hint, the let{s ship this and open an issue to track the upstream fix.

Reply:
Or can we pull in the patches to cobra directly into vendored cobra?

Reply:
You are a master of navigating open-source projects! How did you find that bug at Cobra? Given that this PR at Cobra is hanging around for 13 days already, let's do what you said and ship this. We will pick up this bug fix when Cobra releases a new version and we update dependencies. Not sure we can mess with the `vendor` folder, Go might checksum it. We can also use Cobra's `master` branch for a while. The Git Town test suite is comprehensive enough to create confidence that everything will work.

Reply:
master branch sounds good, I once heard of doing some alternative overlays in the old go packages world, but now with modules there, I'm still a bit confused in general.